### ISSUE TITLE
Bug: guild unable to find model resource

### ISSUE THREAD

**Issue by FelixAbrahamsson** [guildai/guildai#190]
I have an operation called `retrain` that requires the model from a previous run. I found a bug which I think occurs when the hash is on a format that can be converted into a float/integer. I think I remember stumbling upon the same problem in the past, which was fixed in some release, but I think it now occurs again when the hash is long enough. 
I have a run with hash `101e38881c...` which I'm trying to use. If I specify the argument `model=101e38881` or anything shorter than that to my run, guild hangs on 
```
guild: run failed because a dependency was not met: could not resolve 'operation:train|retrain' in model resource: no suitable run for train|retrain
```
However, if I specify `model=101e38881c`, making it not interpretable as a floating number, the run proceeds as normal.

**Comment by gar1t** [comment]
I guess it's a mark of Guild's increasing adoption that we ran into this very special run ID :)

Okay this is fixed in master and will be available in rc10.

Here's the project I used to recreate the condition:

https://github.com/guildai/issue-resolution/tree/master/190-guild-unable-to-find-model-resource

Thanks for the report!

**Comment by gar1t** [comment]
The fix is now generally available in 0.7.0. If this is still an issue, please re-open and we'll sort it out. Thanks again for the report.
